flocker/ca/_ca.py

```diff
@@ -332,6 +355,36 @@ class UserCredential(PRecord):
     username = field(mandatory=True, type=unicode)
 
     @classmethod
+    def from_files(cls, certificate_path, key_path):
```

Comment:
`from_files` can take different file paths for the certificate and the key, `from_path` takes one directory path and expects to find `<username>.crt` and `<username.key>` inside that path - the latter being the original behaviour of the various `Credential` classes, which is still retained because it's still used, but not what we want for `flocker-deploy` anymore. There is a separate issue mentioned in a comment in the code changes in this PR to later and more generally improve the restricted `from_path` behaviour present throughout the credential classes.